packages/nextjs/src/server/clerkCookies.ts

```diff
@@ -0,0 +1,27 @@
+import { constants } from '@clerk/backend';
+
+class ClerkCookies extends Map<string, string> {
+  constructor(cookies: Record<string, string>, namedCookies?: Record<string, string>) {
+    super(Object.entries(cookies));
+    Object.entries(namedCookies || {}).forEach(([k, v]) => {
+      console.log(k, v);
+      Object.defineProperty(this, `get${k}`, { value: () => this.#getNamedCookie(v) });
+    });
+  }
+
+  #getNamedCookie = (name: string) => {
+    return this.get(name);
+  };
+
+  static create<T extends typeof ClerkCookies>(this: T, cookies: Record<string, string>, constants?: Constants) {
+    return new this(cookies, constants) as InstanceType<T> & {
```

Comment:
This is a neat way to dynamically add props to the class instance during runtime and on the type level as well.  Not super happy with the complexity though